packages/wrangler/src/pages/dev.ts

```diff
@@ -703,7 +703,8 @@ export const Handler = async ({
 				.map((d1) => {
 					const { binding, ref } =
 						BINDING_REGEXP.exec(d1.toString())?.groups || {};
-
+					logger.log("binding: ", binding);
+					logger.log("ref: ", ref);
```

Comment:
@mrbbot is it just me or do we never actually use the ref here when creating the local DB?

I've traced it through to here: https://github.com/cloudflare/workers-sdk/blob/ffa01a7d9df25d3ac31e18498659762d67e234f7/packages/miniflare/src/plugins/d1/index.ts#L77C1-L77C1

Reply:
Because if a Pages user supplies a ref via `wrangler pages dev --compatibility-date=2023-09-04 ./public --d1 DB=655e21a7-d6f4-45c4-abdd-46c27591f22a`, I'd expect a call to `durableObjectNamespaceIdFromName('miniflare-D1DatabaseObject', '655e21a7-d6f4-45c4-abdd-46c27591f22a')` rather than `durableObjectNamespaceIdFromName('miniflare-D1DatabaseObject', 'DB')`

Reply:
@GregBrimble / @WalshyDev - I assume that *was* the intention of `--d1 BINDING=ID`, right?

Reply:
`--d1 BINDING=ID` allows users to specify a database ID, which, when provided, will be used for the `.wrangler/state/v3/d1` SQLite file. Ultimately, this lets users work with multi-Worker setups.

I have:

```
package.json
functions/
migrations/
public/
worker/
  package.json
  wrangler.toml
  src/
```

where `/package.json` is:

```json
{
  "scripts": {
    "start:worker": "npm run start --workspace=./worker",
    "start:pages": "npx wrangler pages dev ./public --d1 MY_PAGES_D1_BINDING=my-d1-id",
    "start": "npx concurrently \"npm:start:*\""
  }
}
```

where `/worker/package.json` is:

```
{
  "scripts": {
    "start": "npx wrangler dev --persist-to=../.wrangler/state",
    "migrations:local": "npx wrangler d1 migrations apply MY_WORKER_D1_BINDING --persist-to=../.wrangler/state --local"
  }
}
```

and where `/worker/wrangler.toml` is:

```toml
d1_databases = [
	{ binding = "MY_WORKER_D1_BINDING", database_name = "my-d1-name", database_id = "my-d1-id", migrations_dir = "../migrations" }
]
```

It works reasonably well, but obviously is tricky to set up and isn't well documented at the moment.

I have no idea what `durableObjectNamespaceIdFromName` is or when it should be called.